dashboard/app/models/pd/workshop_daily_survey.rb

```diff
@@ -62,22 +65,47 @@ def set_day_from_form_id
       :day
     )
 
+    LOCAL_CATEGORY = 'local'
+    ACADEMIC_YEAR_CATEGORY = 'academic_year'
+
+    FORM_CATEGORIES = [LOCAL_CATEGORY, ACADEMIC_YEAR_CATEGORY]
+
+    CATEGORY_MAP = {
+      SUBJECT_CSP_SUMMER_WORKSHOP => LOCAL_CATEGORY,
+      SUBJECT_CSP_WORKSHOP_1 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSP_WORKSHOP_2 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSP_WORKSHOP_3 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSP_WORKSHOP_4 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSP_WORKSHOP_5 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSP_WORKSHOP_6 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSP_TEACHER_CON => LOCAL_CATEGORY,
+      SUBJECT_CSD_SUMMER_WORKSHOP => LOCAL_CATEGORY,
+      SUBJECT_CSD_UNITS_2_3 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSD_UNIT_3_4 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSD_UNITS_4_5 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSD_UNIT_6 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSD_UNITS_1_3 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSD_UNITS_4_6 => ACADEMIC_YEAR_CATEGORY,
+      SUBJECT_CSD_TEACHER_CON => LOCAL_CATEGORY,
+    }
     VALID_DAYS = (0..5).freeze
 
     validates_inclusion_of :day, in: VALID_DAYS
 
-    def self.get_form_id_for_day(day)
-      get_form_id 'local', "day_#{day}"
+    def self.get_form_id_for_day(category, day)
```

Comment:
This category is conceptually internal to the survey. It doesn't really make sense to expose and require in a method call. I'd prefer taking a subject here and doing the map internally.